internal/provider/resource_gitlab_project.go

```diff
@@ -1211,7 +1211,11 @@ func resourceGitlabProjectUpdate(ctx context.Context, d *schema.ResourceData, me
 	options := &gitlab.EditProjectOptions{}
```

Comment:
Instead of setting `options.Name` we can already set the `.Name` field during struct initialization:

```suggestion
	options := &gitlab.EditProjectOptions{
		Name: gitlab.String(d.Get("name").(string)),
	}
```

When doing so the check here:

https://github.com/gitlabhq/terraform-provider-gitlab/blob/b5a42dd1612facb1a1a8ea1ef0b9226ec898926c/internal/provider/resource_gitlab_project.go#L1480

will always be `true` - thus, we can remove the check entirely - even if an edit is theoretically not required if the options are equal to only having `Name` set to the already set project name.

Reply:
Ah, this is cleaner. But now I wonder, is this going to cause spurious audit events in Gitlab, where the name is "changed"? We'll want to test for that, I think.